File: src/main/java/toast/pine/Events/PlayerRespondFriendInvite.java
package toast.pine.Events;

import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;
import org.jetbrains.annotations.NotNull;
import toast.pine.SocialSystem.PlayerSocial;

public class PlayerRespondFriendInvite extends Event {

    private static final HandlerList HANDLERS = new HandlerList();

    private final PlayerSocial sender;
    private final PlayerSocial receiver;
    private final Boolean response;

    public PlayerRespondFriendInvite(PlayerSocial sender, PlayerSocial receiver, Boolean response) {
        this.sender = sender;
        this.receiver = receiver;
        this.response = response;
    }

    public PlayerSocial getSender() {
        return this.sender;
    }

    public PlayerSocial getReceiver() {
        return this.receiver;
    }

    public Boolean getResponse() {
        return this.response;
    }


    @Override
    public @NotNull HandlerList getHandlers() {
        return HANDLERS;
    }
}

File: src/main/java/pine/toast/Events/PlayerRespondFriendInvite.kt
package toast.pine.Events

import org.bukkit.event.Event
import org.bukkit.event.HandlerList
import toast.pine.socialSystem.PlayerSocial

class PlayerRespondFriendInvite(sender: PlayerSocial, receiver: PlayerSocial, response: Boolean) : Event() {
    private val sender: PlayerSocial
    private val receiver: PlayerSocial
    private val response: Boolean

    init {
        this.sender = sender
        this.receiver = receiver
        this.response = response
    }

    fun getSender(): PlayerSocial {
        return sender
    }

    fun getReceiver(): PlayerSocial {
        return receiver
    }

    override fun getHandlers(): HandlerList {
        return HANDLERS
    }

    companion object {
        private val HANDLERS: HandlerList = HandlerList()
    }
}
